internal/config/config.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 The Prometheus Authors
```

Comment:
ehm?

Reply:
I just vendored the code since I didn't want the full library.

Reply:
This is why I kept the original copyright notice, instead of just copying the code without credits or asking ChatGPT to write it. https://www.apache.org/licenses/LICENSE-2.0.html


> You must give any other recipients of the Work or Derivative Works a copy of this License; and
> You must cause any modified files to carry prominent notices stating that You changed the files; and
> You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works;
> 
> You may add Your own copyright statement to Your modifications and may provide additional or different license terms and conditions for use, reproduction, or distribution of Your modifications, or for any such Derivative Works as a whole, provided Your use, reproduction, and distribution of the Work otherwise complies with the conditions stated in this License. 

https://www.apache.org/licenses/GPL-compatibility.html

> The Free Software Foundation considers the Apache License, Version 2.0 to be a free software license, compatible with version 3 of the GPL.

Reply:
we're fine then I guess. :+1: